sound/soc/sof/ipc4-pcm.c

```diff
@@ -58,24 +58,23 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		if (!swidget)
 			continue;
 
+		/* find pipeline widget for the pipeline that this widget belongs to */
+		pipeline_widget = swidget->pipe_widget;
+		pipeline = (struct sof_ipc4_pipeline *)pipeline_widget->private;
+
 		/*
 		 * set pipeline state for both FE and BE pipelines for RUNNING state.
 		 * For PAUSE/RESET, set the pipeline state only for the FE pipeline.
 		 */
 		switch (state) {
 		case SOF_IPC4_PIPE_PAUSED:
 		case SOF_IPC4_PIPE_RESET:
-			if (!WIDGET_IS_AIF(swidget->id))
+			if (pipeline->is_backend)
```

Comment:
The logical seems inverted here?

No AIF seems the opposite of a codec-facing pipeline.

Reply:
this code was written originally for passthrough pipelines and for the passthrough pipelines !WIDGET_IS_AIF(swidget->id) would be true for DAI widgets. and a BE pipeline would contain the DAI widget, so the logic is the same right?

Reply:
It depends on the definition of 'AIF'.

Most drivers use 'AIF' for codec-related stuff, e.g.

````
sound/soc/codecs/cs47l15.c:SND_SOC_DAPM_AIF_IN("AIF1RX1", NULL, 0,
sound/soc/qcom/qdsp6/q6afe-dai.c:       SND_SOC_DAPM_AIF_IN("SLIMBUS_4_RX", NULL, 0, SND_SOC_NOPM, 0, 0),
````

assuming that AIF is the contrary to DAI seems far fetched to me.

Reply:
in the SOF topology AIF was used only for host copiers. So Im quite certain this check is correct

Reply:
It seems simply change `!WIDGET_IS_AIF(swidget->id)`  to `WIDGET_IS_DAI(swidget->id)` will work, right?

Reply:
no @bardliao there could be other widgets in the BE pipeline. Take the mixout-gain-dai pipeline for example. WHat happens then?